model-optimizer/extensions/front/tf/ObjectDetectionAPI.py

```diff
@@ -512,19 +515,104 @@ def update_parameter_shape(graph: Graph, match: [SubgraphMatch, None]):
     return initial_input_node_name, parameter_node
 
 
-class ObjectDetectionAPIPreprocessorReplacement(FrontReplacementFromConfigFileSubGraph):
+class ObjectDetectionAPITransformationsStart(FrontReplacementPattern):
     """
-    The class replaces the "Preprocessor" block resizing input image and applying mean/scale values. Only nodes related
-    to applying mean/scaling values are kept.
+    This is a anchor transformation which is used to distinguish TF OD API models related transformations.
     """
-    replacement_id = 'ObjectDetectionAPIPreprocessorReplacement'
+    enabled = True
+
+    def run_after(self):
+        return [CropAndResizeReplacement, FakeQuantWithMinMaxVarsToQuantize]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        pass
```

Comment:
what is a value of this transformation? It is not clear to me how it can distinguish TF OD API models

Reply:
All TF OD API transformations have relationship to the anchor transformations `ObjectDetectionAPITransformationsStart` and `ObjectDetectionAPITransformationsFinish`. Basically all of them are executed after `ObjectDetectionAPITransformationsStart` and before `ObjectDetectionAPITransformationsFinish`. So I grouped all TF OD API transformation into a cluster (refer to the image in the description of the PR). 

TF OD API transformation may leave a graph in a non-normalized state and if some other front transformation will be called in between then this lead to a conversion failure. With the introduction of these anchor transformations we solve this issue and finally we can safely call the graph cleanup to remove dangling nodes after applying all TF OD API transformations after the `ObjectDetectionAPITransformationsFinish`.

We already use similar approach to divide the model conversion pipeline into stages. Refer to the `extensions/front/pass_separator.py` for the example of `FrontStart` and `FrontFinish` anchor transformations.

Reply:
I suggest to add this info into the comment